### PR TITLE
Fix mock pe issues

### PR DESCRIPTION
**Pull request by Kangxun**
Fixes for issues related to filter

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 1]
```diff
@@ -46,6 +46,7 @@ public FilterCommand parse(String args) throws ParseException {
                 category = checkAndParseCategory(secondLastArgument);
             } catch (NumberFormatException e) {
                 // Parse last argument as category
+                checkIfNumberTooLarge(lastArgument);
                 category = checkAndParseCategory(lastArgument);
```
[the remark this answers is not in the document]
Comment (idgrr) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java:50]: The thing that is to be passed in checkAndParseCategory should be the secondlast argument

[the remark this answers is not in the document]
Comment (idgrr) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java:50]: last argument is count. second last is the category

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 2]
```diff
@@ -118,4 +119,19 @@ private Category checkAndParseCategory(String arg) throws ParseException {
...
     }
+
+    /**
+     * Checks if the {@code argument} is a number larger than {@code Integer.MAX_VALUE}.
+     * @throws ParseException if {@code argument} is a number is larger than {@code Integer.MAX_VALUE}
+     */
+    private void checkIfNumberTooLarge(String arg) throws ParseException {
+        try {
+            for (char c : arg.toCharArray()) {
+                Integer.parseInt(String.valueOf(c));
```
[the remark this answers is not in the document]
Comment (Kangxun) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java:130]: Thankss for the suggestion!

[the remark this answers is not in the document]
Comment (idgrr) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java:130]: Ahh I see. So more than maxint is considered an error. In the ug, When u said "If `COUNT` is more than the total number of contacts, all contacts will be shown", I assumed that user will be able to input a super large number (more than maxint) and still be a valid(just show all). I guess u shld either update the ug to reflect that maxint is the upper limit.

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 3]
```diff
@@ -118,4 +119,19 @@ private Category checkAndParseCategory(String arg) throws ParseException {
             throw new ParseException(MESSAGE_INVALID_CATEGORY);
         }
     }
+
+    /**
+     * Checks if the {@code argument} is a number larger than {@code Integer.MAX_VALUE}.
+     * @throws ParseException if {@code argument} is a number is larger than {@code Integer.MAX_VALUE}
+     */
+    private void checkIfNumberTooLarge(String arg) throws ParseException {
```
[the remark this answers is not in the document]
Comment (idgrr) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java:127]: Ohh I see. I think I missed out the part where if not integer, will treat it as category

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kangxun) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java:130]: Ok yep that's a good point will update

**Review by idgrr — APPROVED** [review]
Lgtm

**Pull request merged**